force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,9 +176,34 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        List<Opportunity> opportunities = new List<Opportunity>();
+        List<String> existingOpportunities = new List<String>();
+        Id accountId;
+        for(Opportunity existingOpportunity : [SELECT Name FROM Opportunity WHERE Name IN :oppNames AND Account.Name = :accountName]){
+            existingOpportunities.add(existingOpportunity.Name);
+        }
+        for(Account existingAccount : [SELECT Id FROM Account WHERE Name = :accountName]){
+            accountId = existingAccount.Id;
+        }
+        if(accountId == null){
+            Account newAccount = new Account (Name = accountName);
+            insert newAccount;
```

Comment:
You are now inserting this account in order to call it in the opportunity query which should happen after but it happened already